.github/workflows/ci.yml

```diff
@@ -0,0 +1,197 @@
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+  
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          path: extension
+
+      - name: Checkout Metamod
+        uses: actions/checkout@v4
+        with:
+          repository: alliedmodders/metamod-source
+          ref: ${{ matrix.metamod-version }}
+          path: metamod-source
+          submodules: recursive
+
+      - name: Checkout hl2sdks
+        uses: actions/checkout@v4
+        with:
+          repository: alliedmodders/hl2sdk
+          ref: ${{ matrix.sdks-cs2-version }}
+          path: hl2sdk-${{ matrix.sdks }}
+          submodules: recursive
+
+      - name: Checkout valve protobuf
+        uses: actions/checkout@v4
+        with:
+          repository: SteamDatabase/Protobufs
+          ref: ${{ matrix.protobuf-valve-version }}
+          path: protobuf-valve
+          submodules: recursive
+
+      - name: Checkout protobuf
+        uses: actions/checkout@v4
+        with:
+          repository: protocolbuffers/protobuf
+          ref: ${{ matrix.protobuf-version }}
+          path: protobuf
+          submodules: recursive
+
+      - name: Checkout DynoHook
+        uses: actions/checkout@v4
+        with:
+          repository: qubka/DynoHook
+          ref: ${{ matrix.dynohook-version }}
+          path: DynoHook
+          submodules: recursive
+
+      - name: Install DynoHook
+        run: |
+          sed -i 's/std::string_view name/std::string name/g' DynoHook/src/dynohook/registers.h
+
+          cd DynoHook
+          cmake . -DCMAKE_POSITION_INDEPENDENT_CODE=ON
+          make -j 8
+
+      - name: Install AMBuild
+        run: |
+          python -m pip install wheel
+          pip install git+https://github.com/alliedmodders/ambuild
+
+      - name: Install protobuf
+        shell: bash
+        run: |
+          export EXTENSION_ROOT_DIR=$(pwd)
+          cd protobuf
+
+          cmake . -DCMAKE_POSITION_INDEPENDENT_CODE=ON -Dprotobuf_BUILD_SHARED_LIBS=OFF -DCMAKE_CXX_STANDARD=14 -DCMAKE_INSTALL_PREFIX="$EXTENSION_ROOT_DIR/protobuf"
+          make -j 8
+          make install
+          # Conflict with tier0
+          sed -i 's/const internal::ReflectionSchema\& schema,/const internal::ReflectionSchema\& schemaz,/g' include/google/protobuf/message.h
+          # Conflict with /usr/include/byteswap.h and tier1/byteswap.h
+          sed -i '/#include <byteswap.h>/i \
+          #define bswap_16(x) __bswap_16 (x)\
+          #define bswap_32(x) __bswap_32 (x)\
+          #define bswap_64(x) __bswap_64 (x)' include/google/protobuf/stubs/port.h
+
+          # Compile .proto files to c++
+          cd ../protobuf-valve
+
+          # Counter strike 2
+          ../protobuf/bin/protoc -I=$PWD $PWD/google/protobuf/descriptor.proto --cpp_out=$PWD
+          cd csgo
+          # Conflict variables already defined
+          rm steammessages_base.proto
```

Comment:
not needed anymore, see https://github.com/srcdslab/mm-ext-voice/pull/1/files#r1343034124